Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -267,14 +267,35 @@ extension BrowserViewController: WKNavigationDelegate {
       
       let domainForMainFrame = Domain.getOrCreate(forUrl: mainDocumentURL, persistent: !isPrivateBrowsing)
       
-      // Debouncing logic
+      // Debouncing and query param stripping
       // Handle debouncing for main frame only and only if the site (etld+1) changes
       // We also only handle `http` and `https` requests
       if Preferences.Shields.autoRedirectTrackingURLs.value, url.isWebPage(includeDataURIs: false),
          let currentURL = tab?.webView?.url,
          currentURL.baseDomain != url.baseDomain,
          domainForMainFrame.isShieldExpected(.AdblockAndTp, considerAllShieldsOption: true),
          navigationAction.targetFrame?.isMainFrame == true {
+        // Handle query param stripping
+        if let requestURL = navigationAction.request.url,
```

Comment:
A small optimization you can make here is to check if the `requestURL` has a query string before running through this code. We do this on desktop too: https://github.com/brave/brave-core/blob/7fb458e780ab6165f4a44402b4fe59cbd8647ff4/browser/net/brave_site_hacks_network_delegate_helper.cc#L90-L92